jupyterlab/commands.py

```diff
@@ -90,6 +90,24 @@ def get_app_dir():
     return osp.abspath(app_dir)
 
 
+def dedupe_yarn(path, logger=None):
+    """ `yarn-deduplicate` with the `fewer` strategy to minimize total
+        packages installed in a given staging directory
+
+        This means a extension (or dependency) _could_ cause a downgrade of an
+        version expected at publication time, but core should aggressively set
+        pins above, for example, known-bad versions
+    """
+    had_dupes = Process(
+        ['node', YARN_PATH, 'yarn-deduplicate', '-s', 'fewer', '--fail'],
```

Comment:
> @vidartf Is there a CLI command for going to highest resolved packages for all deps?

We could make this configurable, e.g. `{"LabBuildApp": {"npm_dedupe_strategy": "fewer"}}` (or `highest`) such that there would at least be a knob to fiddle if you ran into issues.

We could also add a `--no-dedupe` flag which would prevent the behavior from happening at all.

Reply:
The issue then becomes applying it consistently in all places:
```yaml
buildutils/src/update-core-mode.ts:  utils.run('jlpm yarn-deduplicate -s fewer --fail', { cwd: staging });
jupyterlab/commands.py:        ['node', YARN_PATH, 'yarn-deduplicate', '-s', 'fewer', '--fail'],
jupyterlab/staging/package.json:    "yarn-deduplicate": "^1.1.1"
package.json:    "build:core": "cd jupyterlab/staging && jlpm && (jlpm yarn-deduplicate -s fewer --fail || jlpm) && jlpm run build",
```
I don't know what to do to get such a config option, once created, into the json and javascript, aside from actually creating a `jupyter lab dedupe` command, which frankly doesn't sound that bad, aside from more things to maintain and document.

If we did want a setting, it seems like we'd need to actually test it.

Reply:
My original quote was not intended to be about the dedupe strategy, but rather about throwing away the known package lock, and getting the latest and greatest (e.g. if you need to pick up a known bug fix).